differs/emerge_diff.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2018 Google, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package differs
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	pkgutil "github.com/GoogleContainerTools/container-diff/pkg/util"
+	"github.com/GoogleContainerTools/container-diff/util"
+	"github.com/sirupsen/logrus"
+)
+
+//Emerge package database location
+const emergePkgFile string = "/var/db/pkg"
+
+type EmergeAnalyzer struct {
+}
+
+func (em EmergeAnalyzer) Name() string {
+	return "EmergeAnalyzer"
+}
+
+// Diff compares the packages installed by emerge.
+func (em EmergeAnalyzer) Diff(image1, image2 pkgutil.Image) (util.Result, error) {
+	diff, err := singleVersionDiff(image1, image2, em)
+	return diff, err
+}
+
+func (em EmergeAnalyzer) Analyze(image pkgutil.Image) (util.Result, error) {
+	analysis, err := singleVersionAnalysis(image, em)
+	return analysis, err
+}
+
+func (em EmergeAnalyzer) getPackages(image pkgutil.Image) (map[string]util.PackageInfo, error) {
+	var path string
+	path = image.FSPath
+	switch path {
+	case "":
+		path = emergePkgFile
+	default:
+		path = filepath.Join(path, emergePkgFile)
+	}
+	packages := make(map[string]util.PackageInfo)
+	if _, err := os.Stat(path); err != nil {
+		// invalid image directory path
+		logrus.Errorf("Invalid image directory path %s", path)
+		return packages, err
+	}
+
+	contents, err := ioutil.ReadDir(path)
+	if err != nil {
+		logrus.Errorf("Non-content in image directory path %s", path)
+		return packages, err
+	}
+
+	for i := 0; i < len(contents); i++ {
```

Comment:
in go, you can iterate directly over the items in a slice:

```suggestion
	for _, content := range contents {
```